packages/schemas/src/types/hook.ts

```diff
@@ -17,6 +19,31 @@ export type HookEventPayload = {
   application?: Pick<Application, 'id' | 'type' | 'name' | 'description'>;
 } & Record<string, unknown>;
 
+export type DataHookEventPayloadWithoutHookId = {
```

Comment:
personally i prefer the opposite way: define `DataHookEventPayload` as a full set, and use `Omit<>` for `DataHookEventPayloadWithoutHookId`

Reply:
As commented above Omit<> will lose type definition on Objects with unknown key values.  https://stackoverflow.com/questions/65013802/loose-type-definition-with-omit-and-keystring-unknown

Reply:
i went through the code, found the "without hook id" types are only used for data constructing. since they are more "business-related", will it be better to keep `schemas` simple, i.e. no "without hook id" types, and use a workaround like:

```ts
type BetterOmit<T, Ignore> = {
  [key in keyof T as key extends Ignore ? never : key]: T[key];
};

type WithoutHookId = BetterOmit<HookEventPayload, 'hookId'>;
```